src/server/uploadServer.ts

```diff
@@ -213,7 +238,18 @@ function uploadAndWatch(
 								if (stats.isDirectory()) {
 									if (action === 'create') {
 										outputChannel.appendLine(`[C ${date}] ${cleanPath(rootDir, fileName)}`);
-										return webdav.mkdir(fileName, rootDir);
+										// we consume the Observable for directory operations to prevent other operations
+										// in a not yet created remote directory
+										return Observable.create(observer => {
+											webdav.mkdir(fileName, rootDir).subscribe(
+												(val) => {
+													observer.next(val);
+													observer.complete;
```

Comment:
Should not be `observer.complete;` called ie  `observer.complete();`?
Also I see no difference in logic between old and new version...

Reply:
also I see no logic for unsubscribe, which will not allow user to cancel operation

Reply:
Good catch with the complete, I'll update that.

Regarding unsubscribe, what should be in there? `Observable.throw(Error('Operation cancelled by user'))`?

Reply:
not really. it waits because complete wasn't called. 

Actually problem here because operations executed in paralell (mkdir and post file is executed is executed together). So all "threads" should stop until mkdir finishes. (currently you are trying to stop only one current "thread" which waits anyway).

> Regarding unsubscribe, what should be in there? 
`webdav.mkdir(fileName, rootDir).subscribe(` returns disposable object so you have to call `.dispose()` once canceling. see reference https://github.com/SqrTT/prophet/blob/42da47d134144d55207e080d5c145696c22f86f4/src/extension.ts#L134

Reply:
actually you find out really interesting problem with currency.

PS. Probably I used wrong term "threads", better term would be a "pool of connections". Uploader tries to upload files concurrently (5 files in 5 connections, see variable CONCURRENT_FILE_UPLOADS).

Reply:
Typically those should be OK, as you can't create a. "foundation" with a file upload. On folders that is a different story and should be serialized. 

This is problematic when scripts create a set of folders, like webpack does in sfra based projects

Reply:
@SGD1953 Maybe can you add the same for deletion. Maybe deletion of folders only if thats possible?

Reply:
I could do that, it won't serve the `CONCURRENT_FILE_UPLOADS` issue though.

Reply:
OK - but as this works for me on webpack, it is defintely a step in the right direction. I'd be happy if you could merge this @SqrTT

Reply:
Hi
I've implemented in a bit different way:

https://github.com/SqrTT/prophet/blob/56dcf9c6e6932cd1d035b0ee572a24fb6a9216f3/src/server/uploadServer.ts#L205-L251

Could you check version 1.2.2 if it solves your issues, please?

Reply:
Not sure I fully understand, might make sense to add some comments (I guess that is generally true as the observable mapping back & forth is very hard to follow).